packages/url_launcher/url_launcher_ios/example/ios/RunnerTests/URLLauncherTests.swift

```diff
@@ -0,0 +1,141 @@
+    launcher = FakeLauncher()
+    plugin = FLTURLLauncherPlugin(launcher: launcher)
+  }
+
+  func testCanLaunchSuccess() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("good://url", error: &error)
+
+    XCTAssertTrue(result!.boolValue)
+    XCTAssertNil(error)
+  }
+
+  func testCanLaunchFailure() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("bad://url", error: &error)
+
+    XCTAssertFalse(result!.boolValue)
+  }
+
+  func testCanLaunchFailureWithInvalidURL() {
+    var error: FlutterError?
+    let result = plugin.canLaunchURL("urls can't have spaces", error: &error)
+
+    XCTAssertNil(result)
+    XCTAssertNotNil(error)
+    XCTAssertEqual(error!.code, "argument_error")
+    XCTAssertEqual(error!.message, "Unable to parse URL")
+    XCTAssertEqual(error!.details as! String, "Provided URL: urls can't have spaces")
+  }
+
+  func testLaunchSuccess() {
+    let expectation = XCTestExpectation(description: "completion called")
+    plugin.launchURL("good://url", universalLinksOnly: false) { result, error in
+      XCTAssertTrue(result!.boolValue)
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+
+    wait(for: [expectation], timeout: 1)
+  }
+
+  func testLaunchFailure() {
+    let expectation = XCTestExpectation(description: "completion called")
+
+    plugin.launchURL("bad://url", universalLinksOnly: false) { result, error in
+      XCTAssertFalse(result!.boolValue)
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+
+    wait(for: [expectation], timeout: 1)
+  }
+
+  func testLaunchFailureWithInvalidURL() {
+    let expectation = XCTestExpectation(description: "completion called")
+
+    plugin.launchURL("urls can't have spaces", universalLinksOnly: false) { result, error in
+      XCTAssertNil(result)
+      XCTAssertNotNil(error)
+      XCTAssertEqual(error!.code, "argument_error")
+      XCTAssertEqual(error!.message, "Unable to parse URL")
+      XCTAssertEqual(error!.details as! String, "Provided URL: urls can't have spaces")
+
+      expectation.fulfill()
+    }
+
+    wait(for: [expectation], timeout: 1)
+  }
+
+  func testLaunchWithoutUniversalLinks() {
+    let expectation = XCTestExpectation(description: "completion called")
+    plugin.launchURL("good://url", universalLinksOnly: false) { result, error in
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+
+    wait(for: [expectation], timeout: 1)
+
+    XCTAssertEqual(launcher.passedOptions?[.universalLinksOnly] as? Bool, false)
+  }
+
+  func testLaunchWithUniversalLinks() {
+    let expectation = XCTestExpectation(description: "completion called")
+
+    plugin.launchURL("good://url", universalLinksOnly: true) { result, error in
+      XCTAssertNil(error)
+      expectation.fulfill()
+    }
+
+    wait(for: [expectation], timeout: 1)
+
+    XCTAssertEqual(launcher.passedOptions?[.universalLinksOnly] as? Bool, true)
+  }
+
+}
+
+final fileprivate class FakeLauncher: NSObject, FULLauncher {
```

Comment:
Do we actually need this to inherit from NSObject?

Reply:
yes since the plugin is still in objective-c